scaleway/resource_rdb_instance_beta.go

```diff
@@ -0,0 +1,206 @@
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Name of the database instance",
+			},
+			"node_type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The type of database instance you want to create",
+			},
+			"engine": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Database's engine version id",
+			},
+			"is_ha_cluster": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
+				Default:     false,
+				Description: "Enable or disable high availability for the database instance",
+			},
+			"disable_backup": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Disable automated backup for the database instance",
+			},
+			"user_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Identifier for the first user of the database instance",
+			},
+			"password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Password for the first user of the database instance",
+			},
+			"tags": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "List of tags [\"tag1\", \"tag2\", ...] attached to a database instance",
+			},
+
+			// TODO: computed (endpoint_ip,endpoint_port,read_replicas,certificate,backup_schedule)
+
+			"region":          regionSchema(),
+			"organization_id": organizationIDSchema(),
+		},
+	}
+}
+
+func resourceScalewayRdbInstanceBetaCreate(d *schema.ResourceData, m interface{}) error {
+	rdbAPI, region, err := getRdbAPIWithRegion(d, m)
+	if err != nil {
+		return err
+	}
+
+	createReq := &rdb.CreateInstanceRequest{
+		Region:         region,
+		OrganizationID: d.Get("organization_id").(string),
+		Name:           expandOrGenerateString(d.Get("name"), "rdb"),
+		NodeType:       d.Get("node_type").(string),
+		Engine:         d.Get("engine").(string),
+		IsHaCluster:    d.Get("is_ha_cluster").(bool),
+		DisableBackup:  d.Get("disable_backup").(bool),
+		UserName:       d.Get("user_name").(string),
+		Password:       d.Get("password").(string),
+		Tags:           expandStrings(d.Get("tags")),
+	}
+
+	res, err := rdbAPI.CreateInstance(createReq)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(newRegionalId(region, res.ID))
+
+	_, err = rdbAPI.WaitForInstance(&rdb.WaitForInstanceRequest{
+		Region:     region,
+		InstanceID: res.ID,
+		Timeout:    InstanceServerWaitForTimeout,
+	})
+	if err != nil {
+		return err
+	}
+
+	return resourceScalewayRdbInstanceBetaRead(d, m)
+}
+
+func resourceScalewayRdbInstanceBetaRead(d *schema.ResourceData, m interface{}) error {
```

Comment:
what is `m`?

Reply:
m is the common terraform variable for `meta` https://www.terraform.io/docs/extend/writing-custom-providers.html#defining-resources